scrape/scrape.go

```diff
@@ -1034,6 +1039,16 @@ func (sl *scrapeLoop) scrapeAndReport(interval, timeout time.Duration, last time
 		)
 	}
 
+	// Temporary workaround for a jitter in go timers that causes disk space
+	// increase in TSDB.
+	// See https://github.com/prometheus/prometheus/issues/7846
+	if last.IsZero() && interval > 100*scrapeTimestampTolerance {
```

Comment:
Why do we care about last, which is when the last scrape happened? We should care about when it should have happened.

Reply:
It would never "fix" the timestamp when we skip a scrape.

Reply:
I'm not following, we should still be working off when the last scrape should have happened. Not some other number that may be different by up to the scrape interval.

Reply:
I have adapted the code to be closer to that idea.